### PR TITLE
Add peering support

### PR DESCRIPTION
**Pull request by aschmahmann**
While this shouldn't be strictly needed in production it's both available for us if we want it (e.g. for high value connections we explicitly want to protect) and it can be of use during testing (e.g. #34).

IIRC @hsanjuan also requested this

### CODE REVIEW

**Review by hacdias — APPROVED** [review]

File: setup.go [hunk 1]
```diff
@@ -269,6 +271,14 @@ func Setup(ctx context.Context, cfg Config, key crypto.PrivKey, dnsCache *cached
 		return nil, err
 	}
 
+	ps := peering.NewPeeringService(h)
+	if err := ps.Start(); err != nil {
+		return nil, err
+	}
+	for _, a := range cfg.Peering {
+		ps.AddPeer(a)
+	}
```
[the remark this answers is not in the document]
Comment (aschmahmann) [reply setup.go:280]: It does very little work without any peers, and I don't recall it ever being an issue in kubo (which runs it by default) even where there are no peering peers registered.

It's fine to fix though. I'll bump go-libp2p along with it https://github.com/libp2p/go-libp2p/releases/tag/v0.32.1

**Comment by acejam** [comment]
@aschmahmann Is there a way to list the current peers that `rainbow` is connected to? I'm trying to verify that this feature works.

**Comment by aschmahmann** [comment]
No, there isn't an RPC call that lists the peers the node is connected to. You may be able to detect the go-libp2p connect/disconnect events by looking through the logs though if you're just doing it for debugging. You could also check on the peer you're supposed to be connecting to if they're connected (peering should be symmetrical for stability purposes anyway).

Note: if you're running with two libp2p hosts logs will be from both hosts so might be confusing. If you want different Amino DHT + Bitswap hosts but not in the same binary you can choose to run the Amino DHT as a separate binary (e.g. a non-publicly exposed kubo node or https://github.com/ipfs-shipyard/someguy) and hook it up to rainbow via the routing-v1 API (supported in rainbow).

**Comment by acejam** [comment]
> No, there isn't an RPC call that lists the peers the node is connected to. You may be able to detect the go-libp2p connect/disconnect events by looking through the logs though if you're just doing it for debugging. You could also check on the peer you're supposed to be connecting to if they're connected (peering should be symmetrical for stability purposes anyway).
> 
> Note: if you're running with two libp2p hosts logs will be from both hosts so might be confusing. If you want different Amino DHT + Bitswap hosts but not in the same binary you can choose to run the Amino DHT as a separate binary (e.g. a non-publicly exposed kubo node or https://github.com/ipfs-shipyard/someguy) and hook it up to rainbow via the routing-v1 API (supported in rainbow).

Logs work for me. Whats the best way to set the different subsystem logs to debug in `rainbow`? I've only used `ipfs log level` with `kubo` before.

**Pull request merged**